eth-bytecode-db/eth-bytecode-db/src/verification/db.rs

```diff
@@ -78,39 +89,12 @@ pub(crate) async fn insert_verified_contract_data(
     Ok(())
 }
 
-async fn insert_source_details(
-    txn: &DatabaseTransaction,
-    source: types::Source,
-) -> Result<sources::Model, anyhow::Error> {
-    let abi = match source.abi {
-        None => None,
-        Some(abi) => serde_json::from_str(&abi).context("deserialize abi")?,
-    };
-    let source = sources::ActiveModel {
-        source_type: Set(source.source_type.into()),
-        compiler_version: Set(source.compiler_version),
-        compiler_settings: Set(serde_json::from_str(&source.compiler_settings)
-            .context("deserialize compiler settings")?),
-        file_name: Set(source.file_name),
-        contract_name: Set(source.contract_name),
-        raw_creation_input: Set(source.raw_creation_input),
-        raw_deployed_bytecode: Set(source.raw_deployed_bytecode),
-        abi: Set(abi),
-        ..Default::default()
-    }
-    .insert(txn)
-    .await
-    .context("insert into \"sources\"")?;
-
-    Ok(source)
-}
-
-async fn insert_source_files(
+async fn insert_files(
     txn: &DatabaseTransaction,
-    source_model: &sources::Model,
-    source_files: BTreeMap<String, String>,
-) -> Result<(), anyhow::Error> {
-    for (name, content) in source_files {
+    files: BTreeMap<String, String>,
+) -> Result<Vec<files::Model>, anyhow::Error> {
+    let mut result = Vec::new();
+    for (name, content) in files {
         let file = {
             let file = files::Entity::find()
```

Comment:
may be (later?) make this concurent

Reply:
Need to think about that, but for now it seems as a nice idea